runtime/shared_common/CacheMap.cpp

```diff
@@ -229,20 +229,31 @@ SH_CacheMap::newInstance(J9JavaVM* vm, J9SharedClassConfig* sharedClassConfig, S
  * the metadata which have been accessed to date.
  */
 void
-SH_CacheMap::dontNeedMetadata(J9VMThread* currentThread) 
+SH_CacheMap::dontNeedMetadata(J9VMThread* currentThread)
 {
 	Trc_SHR_CM_j9shr_dontNeedMetadata(currentThread);
-	SH_CompositeCacheImpl* ccToUse = _ccHead;
 
 	if (_metadataReleased) {
 		return;
 	}
+	SH_CacheMap::forceDontNeedMetadata(currentThread);
+}
+
+/**
+ * Advise the OS to release resources associated with
+ * the metadata which have been accessed to date, regardless of
+ * whether or not this advice was issued before.
+ */
+void
+SH_CacheMap::forceDontNeedMetadata(J9VMThread* currentThread)
+{
+	SH_CompositeCacheImpl* ccToUse = _ccHead;
```

Comment:
We can add a tracepoint to this method, similar to Trc_SHR_CM_j9shr_dontNeedMetadata